preview/interfaces/Geometry.cpp

```diff
@@ -522,6 +522,59 @@ namespace infra
         return !(*this == other);
     }
 
+    RowFirstPoints::RowFirstPoints(infra::Region region)
+        : region(region)
+        , current(region.TopLeft())
+    {}
+
+    RowFirstPoints RowFirstPoints::begin() const
+    {
+        return *this;
+    }
+
+    RowFirstPoints RowFirstPoints::end() const
+    {
+        RowFirstPoints result(*this);
+        result.current = region.BottomRight();
+        result.current.x = region.Left();
+        return result;
+    }
+
+    Point RowFirstPoints::operator*() const
+    {
+        return current;
```

Comment:
This function feels a bit misleading, have you considered a function like "Point()" instead?

Reply:
The purpose of RowFirstPoints is to be used in for-loops; there it has to behave like an iterator, so it has to have operator*